### DISCUSSION TITLE
`languages.toml` giving error after building 23.05 from source

### DISCUSSION THREAD

**Discussion by sourproton** [helix-editor/helix#7107] [General]
I have the following `languages.toml`:

```
[[language]]
name = "python"
roots = ["pyproject.toml"]

[language.language-server]
command = "pyright-langserver"
args = ["--stdio"]

[language.config] # <- this is the important line


[[language]]
name = "rust"

[language.config]
check = { command = "clippy" }
```

, which worked fine in version 23.03.

After updating the helix directory with `git pull` and building from source with `cargo install --path helix-term --locked` and `ln -s $PWD/runtime ~/.config/helix/runtime`, I have the following error when launching helix:

```
Bad language config: unknown field `config`, expected one of `name`, `language-id`, `scope`, `file-types`, `shebangs`, `roots`, `comment-token`, `text-width`, `soft-wrap`, `auto-format`, `formatter`, `diagnostic-severity`, `grammar`, `injection-regex`, `language-servers`, `indent`, `debugger`, `auto-pairs`, `rulers`, `workspace-lsp-roots`
in `language`
```

**Comment by pascalkuthe** [comment]
There were breaking changes recently in the master branch. Check the master docs

**Reply by sourproton** [reply]
Thank you for pointing this out. I assumed the instructions in the installation page of the regular docs were for installing the latest stable, not master. What is the correct way of installing the stable branch then? `git clone` then `git checkout tags/23.05` before `cargo install`?

**Comment by quomat** [comment]
Note for the future reference: The `config` field was moved to (in this case) `[language-server.pyright-langserver]`. In general, `[language-server.$name]`.

So to repair `clippy` you just need to change 
```
[language.config]
check = { command = "clippy" }
```
to
```
[language-server.rust-analyzer.config]
checkOnSave = { command = "clippy" }
```

I don't know why, but only `checkOnSave` works for me here, even though the [rust-analyzer documentation](https://rust-analyzer.github.io/manual.html) doesn't even say that `checkOnSave` has a `command` property, so you can try both `checkOnSave` as well as `check`.
